### ISSUE TITLE
9 - Forum

### ISSUE THREAD

**Issue by bethlakshmi** [bethlakshmi/GBE2#389]
**First part of the revised ticket is get a Forum up and running on the Test server, merged into the codebase, with a way to turn it off**

For ad hoc communication between community members.  Preferably a forum in a can that we can stage and only moderate, not bug fix.

This would require a login and password, ideally it would utilize the same login and password that people with accounts already use. It would be great if once you were logged into the overall site you didn't need to login to the forum again.

Schedule:
April 1 - April 30: Review available existing Django software solutions
May 1 - May 31: Installation and testing; learning how to be a moderator; writing any documentation needed for *our* users
June 1: live on the site

-----------------

Add a discussion forum where registered users can find roommates, ride shares, and generally chat about the event.

Something like http://flaskbb.org/ -- no point in reinventing the wheel.

Would require looking at their registration system and figuring out how to let our registered users participate in the forum without registering twice (once for GBE site, once for forum)

Note for @burlexpo -- there is also http://www.websitetoolbox.com/forum-hosting/index.html if for some reason we can't do this in-house.

**Comment by burlexpo** [comment]
I'd like this up and running before we start accepting bids for classes and and acts.  People should be able to use it sooner rather than later; deadline for full functional integration is June 1.

**Comment by burlexpo** [comment]
Again, assigning this to @jonkiparsky because I'd love to see some new functionality on the site and I've been asking for this particular functionality for a while... ;-)

**Comment by BenignCremator** [comment]
Have gotten an install of DjangoBB-Forum working in parallel with GBE, using the GBE user registrations and information.  It is NOT a clean install, and some things are broken (user lists and profiles, in particular), but the main forum functions all work.  Currently editing the Vagrantfile to get a mostly automatted install, which I will use for real integration and testing.  Will submit that as branch GBE-Forum, due to a branch already named GBE-389 existing.  Claiming this issue ticket and removing its backlog status.

**Comment by jonkiparsky** [comment]
Use git -D GBE-xxx to delete a branch.

sent from my handy. typos are just part of the fun.

**Comment by BenignCremator** [comment]
On 11/15/2015 06:48 PM, Jon Kiparsky wrote:
> Use git -D GBE-xxx to delete a branch.

Will do.  The branch will become GBE-389.  The previous GBE-389 will be 
gone soon, and I will update the cron job files to a different branch.

Hunter

**Comment by BenignCremator** [comment]
Have a 2nd pass at an install via a Vagrant file, breaks pagination tags, so posting is broken.  Was working on the hand tuned install, but have not address config for pagination and some of the other minor components yet.

**Comment by BenignCremator** [comment]
Have the Vagrant file more or less sorted out.  Vagrant up now installs everything needed, is a fairly clean fashion, and has the settings for the major features to work.  Have tested creating, deleting, and editing categories, forums, and posts, both as GBEAdmin and a regular user.  Feel like I am ready to put this on gbetest for other people to look at it, but wondering if we need anti-spam features first.  Remaining blocking issues for live are captcha/anti-spam features, and major issues are skinning the forum to look like the rest of the site (I am ok with it going live with an ugly skin, and skinning it later) and user profiles.  User lists works, but only returns a list of that users posts, not their profile.

**Comment by jonkiparsky** [comment]
I've checked out GBE-389 and from looking at the urlconf I see that /forum is the place to go, so I can get there
However, I can't really see much working. Can't find a way to create posts or topics or anything, search by users returns no users (even the user that I'm signed in as) and the "Profile" link (http://localhost:2200/forum/user/jpk/) gets an error. 

I think there's a good bit of work left to do here.

**Comment by BenignCremator** [comment]
On 11/19/2015 03:53 PM, Jon Kiparsky wrote:

> I've checked out GBE-389 and from looking at the urlconf I see that
> /forum is the place to go, so I can get there
> However, I can't really see much working. Can't find a way to create
> posts or topics or anything, search by users returns no users (even the
> user that I'm signed in as) and the "Profile" link
> (http://localhost:2200/forum/user/jpk/) gets an error.

> I think there's a good bit of work left to do here.

You have to create categories and forums through the admin interface. 
After that, posting topics and comments is easy.  Creating categories 
and forum through admin is, I admit, odd, but understandable, since only 
admins will do it.  So first you need to create a category for a forum 
to reside in:

http://localhost:8282/admin/djangobb_forum/category/add/

Then you need to create a forum for people to post topics and comments into:

http://localhost:8282/admin/djangobb_forum/forum/add/

Search by user, as stated, returns the forum topics and comments from 
that user.  Seems odd, but that is what it does, and you need to have 
content in the forums for something to be returned.  Since you have 
nothing, you got nothing.  It is not a general site user search, but 
forum content search with an odd label.

And, yes, I stated that the profile link does not work.  Part of that is 
that the GBE profiles do not quite work as other Django apps expect, due 
to questions about the public and private persona.  Betty and I talked 
about it, and the conclusion was that it should be broken right now, and 
we should comment out that link until we get profiles working better 
elsewhere on the site.  This will probably be GBE catching URLs to 
forum/profile/* and redirecting them elsewhere.

And, yes, there is more work to do, but not a ton that is blocking, and 
a good bit that is initial content or anti-spam security.  We need a 
login dependent link on the main menu, so only registered and confirmed 
users can see that we have a forum, enable captcha on the forum for 
first posts or user registration or something, and add our banner and 
menu at the top of the page.  So I am ok with going live with an ugly 
forum and adding a skin to match the rest of the site, and fixing the 
user profile problems later.

Right now, I am fighting with gbetest.  Pip does not seem to be working 
there, so I am not able to install the dependencies.  Once I get over 
that, I will set up some initial content, so everyone can see how it works.

Hunter

**Comment by jonkiparsky** [comment]
Why is this code in our codebase? We should be installing it through settings.INSTALLED_APPS, not including it in our repo.

**Comment by BenignCremator** [comment]
On 11/19/2015 07:33 PM, Jon Kiparsky wrote:

> Why is this code in our codebase? We should be installing it through
> settings.INSTALLED_APPS, not including it in our repo.

A mistake.  Only the part that has the settings and scripting should be 
in the branch for this ticket.  The rest should not be there.  From 
there, it should go into the wiki, not master.

And I just double checked.  None of their code is in our codebase.  Only 
the settings and scripts I wrote that are needed to bring it up.  It is 
a different Vagrantfile, urls.py,

Hunter

**Comment by BenignCremator** [comment]
Have reworked with setup of DjangoBB_Forums, so that it works with the current code base.  Have also modified the settings.py, urls.py, and a couple of other things to get it to have an off option from local_settings.py.  Works with an older version of DjangoBB_Forums, as newer versions required Django >=1.7.  Going to test more tomorrow before issuing a PR.

**Comment by BenignCremator** [comment]
Final commits, and submitting for PR.

**Comment by BenignCremator** [comment]
Ok, too much water under the bridge.  Spent three hours trying to sort out the changes between this, the original version of it, the old master circa Sept 2015, and current master, and decided that it was not worth risk restoring antebellum bugs into master because I could not keep the diffs straight in my head on >200 files.  Wiped out the PR and old branch, rebranched from master, redid the work from backups, retested, and reissuing the PR.  Whee!
